yaml/_type.ts

```diff
@@ -10,44 +10,15 @@ export type KindType = "sequence" | "scalar" | "mapping";
 export type StyleVariant = "lowercase" | "uppercase" | "camelcase" | "decimal";
 export type RepresentFn = (data: Any, style?: StyleVariant) => Any;
 
-interface TypeOptions {
-  kind: KindType;
-  resolve?: (data: Any) => boolean;
-  construct?: (data: string) => Any;
-  instanceOf?: Any;
-  predicate?: (data: Record<string, unknown>) => boolean;
-  represent?: RepresentFn | ArrayObject<RepresentFn>;
-  defaultStyle?: StyleVariant;
-  styleAliases?: ArrayObject;
-}
-
-function checkTagFormat(tag: string): string {
-  return tag;
-}
-
-export class Type {
+export interface Type {
   tag: string;
-  kind: KindType | null = null;
-  instanceOf: Any;
+  kind: KindType | null;
+  instanceOf?: Any;
   predicate?: (data: Record<string, unknown>) => boolean;
   represent?: RepresentFn | ArrayObject<RepresentFn>;
   defaultStyle?: StyleVariant;
   styleAliases?: ArrayObject;
   loadKind?: KindType;
-
-  constructor(tag: string, options?: TypeOptions) {
-    this.tag = checkTagFormat(tag);
-    if (options) {
-      this.kind = options.kind;
-      this.resolve = options.resolve || (() => true);
-      this.construct = options.construct || ((data: Any): Any => data);
-      this.instanceOf = options.instanceOf;
-      this.predicate = options.predicate;
-      this.represent = options.represent;
-      this.defaultStyle = options.defaultStyle;
-      this.styleAliases = options.styleAliases;
-    }
-  }
-  resolve: (data?: Any) => boolean = (): boolean => true;
-  construct: (data?: Any) => Any = (data): Any => data;
+  resolve?: (data?: Any) => boolean;
```

Comment:
Making resolve optional feels strange to me

Reply:
> An alternative would be to require resolve and construct methods in Type and add them to types that are missing them.

I think that is more correct way to perform this transition.

Reply:
this line https://github.com/denoland/deno_std/pull/5262/files#r1663457908 seems a bug caused by the optionality of the field